Modules/_testcapimodule.c

```diff
@@ -2335,6 +2335,80 @@ get_timezone_utc_capi(PyObject* self, PyObject *args) {
     }
 }
 
+static PyObject *
+get_date_fromtimestamp(PyObject* self, PyObject *args) {
+    int ts;
+    int macro = 0;
+    if (!PyArg_ParseTuple(args, "i|p", &ts, &macro)) {
```

Comment:
No need to convert timestamp to C long. Use "O" instead of "i". And you can use `PyTuple_Pack()` for creating an arguments tuple.